opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -352,19 +346,31 @@ def __init__(
         start_time: Optional[int] = None,
         end_time: Optional[int] = None,
     ) -> None:
+        super().__init__(attributes)
         self._name = name
         self._context = context
         self._kind = kind
         self._instrumentation_info = instrumentation_info
         self._parent = parent
         self._start_time = start_time
         self._end_time = end_time
-        self._attributes = attributes
         self._events = events
         self._links = links
         self._resource = resource
         self._status = status
 
+    @property
+    def dropped_events(self) -> int:
+        if self._events:
```

Comment:
Is this check really necessary?

Reply:
`_events` could be None so I would say yes, but happy to hear your thoughts.